### PR TITLE
Qdrant v1 implementation

### LINKED ISSUE #5172: [Qdrant] v1 implementation of DustQdrantClient methods

### PR DESCRIPTION
**Pull request by spolu**
## Description

- Introduce `cluster-0`
- Bumps qdrant lib to 1.9
- Implements v1 (sharding + multi-tenancy)

Fixes https://github.com/dust-tt/dust/issues/5172

## Risk

N/A not in prod but will ensure tenant separation, so review with care.

## Deploy Plan

- N/A

### CODE REVIEW

File: core/src/data_sources/qdrant.rs [hunk 1]
```diff
@@ -195,24 +205,54 @@ impl DustQdrantClient {
...
+                )
             }
         }
     }
 
+    fn shard_key(&self, data_source: &DataSource) -> shard_key::Key {
+        // We use the last character of the internal_id to determine the pool_id. This id is
+        // generated using new_id and is guaranteed random. Using the last character gives us a
+        // path to moving data sources across shard when needed.
+        let pool_id = data_source.internal_id().chars().last().unwrap() as u8 % SHARD_KEY_COUNT;
```
Comment (flvndvd) [inline core/src/data_sources/qdrant.rs:228]: Why pool_id?

File: core/src/data_sources/qdrant.rs [hunk 2]
```diff
@@ -195,24 +205,54 @@ impl DustQdrantClient {
...
+        format!("pool_{}", pool_id).into()
+    }
+
+    // Inject the `data_source_internal_id` to the filter to ensure tenant separation. This
+    // implementaiton ensure data separation of our users data. Modify with extreme caution.
+    fn enforce_tenant(&self, data_source: &DataSource, filter: &mut qdrant::Filter) -> () {
```
Comment (flvndvd) [inline core/src/data_sources/qdrant.rs:234]: What about `apply_tenant_filter`?

**Review by flvndvd — APPROVED** [review]
LGTM, left some small nits.

File: core/src/data_sources/qdrant.rs (hunk 2 above)
Comment (spolu) [reply core/src/data_sources/qdrant.rs:234]: No strong feeling. Feels less scary though.

Comment (spolu) [reply core/src/data_sources/qdrant.rs]: Udpated :+1:

File: core/src/data_sources/qdrant.rs [hunk 3]
```diff
@@ -311,15 +349,24 @@ impl DustQdrantClient {
...
-                // v0 implementation
                 self.client
                     .delete_collection(self.collection_name(data_source))
                     .await?;
             }
             QdrantClusterVersion::V1 => {
-                // TODO: v1 implementation
-                // - v1 implementation will delete points not collection.
-                unimplemented!()
+                // Create a default filter and ensure teanant separation to delete all the points
```
[the remark this answers is not in the document]
Comment (spolu) [reply core/src/data_sources/qdrant.rs]: :+1:

**Pull request merged**